### PR TITLE
Add Pi 5 workaround for w1-gpio

### PR DESCRIPTION
**Pull request by P33M**
Slightly less hacky - parse that we want read-polling out of devicetree, so it's used only where needed.

### CODE REVIEW

File: arch/arm/boot/dts/overlays/w1-gpio-pullup-pi5-overlay.dts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Definitions for w1-gpio module (without external pullup)
+/dts-v1/;
+/plugin/;
+
+#include "w1-gpio-overlay.dts"
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/w1-gpio-pullup-pi5-overlay.dts]: `#include "w1-gpio-pullup-overlay.dts"`

File: arch/arm/boot/dts/overlays/overlay_map.dts [hunk 2]
```diff
@@ -450,4 +450,13 @@
 	vl805 {
 		bcm2711;
 	};
+
+	w1-gpio {
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/overlay_map.dts:464]: An omitted platform is an unsupported platform, so both non-pi5 overlays also need:
```
		bcm2835;
		bcm2711;
```

File: arch/arm/boot/dts/overlays/w1-gpio-pullup-overlay.dts [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 	fragment@0 {
 		target-path = "/";
-		__overlay__ {
+		frag0: __overlay__ {
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/w1-gpio-pullup-overlay.dts]: Not needed.

File: arch/arm/boot/dts/overlays/w1-gpio-pullup-pi5-overlay.dts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+/ {
+	compatible = "brcm,bcm2712";
+
+	fragment@2 {
+		target = <&frag0>;
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/w1-gpio-pullup-pi5-overlay.dts]: `target = <&w1>;`

File: arch/arm/boot/dts/overlays/w1-gpio-overlay.dts [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
 	fragment@0 {
 		target-path = "/";
-		__overlay__ {
+		frag0: __overlay__ {
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/w1-gpio-overlay.dts]: Not required.

File: arch/arm/boot/dts/overlays/w1-gpio-pi5-overlay.dts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+/dts-v1/;
+/plugin/;
+
+#include "w1-gpio-overlay.dts"
+
+/ {
+	compatible = "brcm,bcm2712";
+
+	fragment@2 {
+		target = <&frag0>;
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/w1-gpio-pi5-overlay.dts]: `target = <&w1>;`

File: arch/arm/boot/dts/overlays/w1-gpio-pullup-overlay.dts (hunk 3 above)
Comment (P33M) [reply arch/arm/boot/dts/overlays/w1-gpio-pullup-overlay.dts]: Arg, missed this one - one moment please...

File: arch/arm/boot/dts/overlays/overlay_map.dts (hunk 2 above)
Comment (pelwell) [reply arch/arm/boot/dts/overlays/overlay_map.dts:464]: Sorry - I must have imagined these were already there: Can you add them?:
```
       w1-gpio-pi5 {
               bcm2712;
       };

       w1-gpio-pullup-pi5 {
               bcm2712;
       };
```

**Pull request merged**